src/app/debug/table/table-settings-modal/table-settings-modal.component.ts

```diff
@@ -155,25 +156,26 @@ export class TableSettingsModalComponent implements OnDestroy {
     this.settingsForm.reset();
     this.settingsService.setShowMultipleAtATime();
     this.httpService.resetSettings().subscribe({
-      next: (response) => this.saveResponseSetting(response),
+      next: (response: OptionsSettings) => this.saveResponseSetting(response),
       error: () => catchError(this.errorHandler.handleError()),
     });
     this.httpService.getTransformation(true).subscribe({
-      next: (res) => this.settingsForm.get('transformation')?.setValue(res.transformation),
+      next: (res: Record<string, string>) => this.settingsForm.get('transformation')?.setValue(res.transformation),
       error: () => catchError(this.errorHandler.handleError()),
     });
   }
 
   loadSettings(): void {
     this.httpService.getSettings().subscribe({
-      next: (response) => this.saveResponseSetting(response),
+      next: (response: OptionsSettings) => this.saveResponseSetting(response),
       error: () => catchError(this.errorHandler.handleError()),
     });
     if (localStorage.getItem('transformationEnabled')) {
       this.settingsForm.get('transformationEnabled')?.setValue(localStorage.getItem('transformationEnabled') == 'true');
     }
     this.httpService.getTransformation(false).subscribe({
-      next: (response) => this.settingsForm.get('transformation')?.setValue(response.transformation),
+      next: (response: Record<string, string>) =>
+        this.settingsForm.get('transformation')?.setValue(response.transformation),
```

Comment:
response is of type Record<string, string>, but you are able to do response.transformation? 
Should that response have a type then?